base-theme/assets/js/hooks/userLists/index.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  useInfiniteQuery,
+  useMutation,
+  useQuery,
+  useQueryClient
+} from "@tanstack/react-query"
+import { userListsApi } from "../../clients"
+import type {
+  UserlistsApiUserlistsListRequest as ListRequest,
+  UserlistsApiUserlistsCreateRequest as CreateRequest,
+  UserlistsApiUserlistsDestroyRequest as DestroyRequest,
+  UserlistsApiUserlistsItemsListRequest as ItemsListRequest,
+  UserList
+} from "@mitodl/open-api-axios/v1"
+import { userlistKeys, userlistQueries } from "./queries"
+import { useUserIsAuthenticated } from "../user"
+
+const useUserListList = (
+  params: ListRequest = {},
+  opts?: { enabled?: boolean }
+) => {
+  const queryOptions = userlistQueries.list(params)
+  return useQuery({
+    ...queryOptions,
+    queryKey: queryOptions.queryKey ?? [`userlist-list`, params],
+    ...opts
+  })
```

Comment:
I was confused why you added queryKey explicitly here, but saw you got a TS error otherwise.

If you define `userlistQueries` using the queryOptions helper from `@tanstack/react-query`, you will no longer get a TS error if you omit the query key.

BUT my real rec would be to not both defining these custom query hooks.